routing.rst

```diff
@@ -116,57 +86,127 @@ Now you can configure the routes:
+             */
+            public function show(int $id)
+            {
+                // ... return a JSON response with the post
+            }
+
+            /**
+             * @Route("/api/posts/{id}", methods={"PUT"})
+             */
+            public function edit(int $id)
+            {
+                // ... edit a post
+            }
+        }
+
+    .. code-block:: yaml
+
+        # config/routes.yaml
+        api_post_show:
+            path:       /api/posts/{id}
+            controller: App\Controller\BlogApiController::show
+            methods:    GET|HEAD
+
+        api_post_edit:
+            path:       /api/posts/{id}
+            controller: App\Controller\BlogApiController::edit
+            methods:    PUT
+
+    .. code-block:: xml
+
+        <!-- config/routes.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                https://symfony.com/schema/routing/routing-1.0.xsd">
 
-    However, the slash ``/`` is ignored by default in placeholder values because
-    the router uses it as separator between different placeholders.
-    To learn more about this, you can read
-    :ref:`routing/slash_in_parameter`.
+            <route id="api_post_show" path="/api/posts/{id}"
+                controller="App\Controller\BlogApiController::show"
+                methods="GET|HEAD"/>
 
-Each route also has an internal name: ``blog_list`` and ``blog_show``. These can
-be anything (as long as each is unique) and don't have any meaning yet. You'll
-use them later to :ref:`generate URLs <routing-generate>`.
+            <route id="api_post_edit" path="/api/posts/{id}"
+                controller="App\Controller\BlogApiController::edit"
+                methods="PUT"/>
+        </routes>
+
+    .. code-block:: php
+
+        // config/routes.php
+        use App\Controller\BlogApiController;
+        use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->add('api_post_show', '/api/posts/{id}')
+                ->controller([BlogApiController::class, 'show'])
+                ->methods(['GET', 'HEAD'])
+            ;
+            $routes->add('api_post_edit', '/api/posts/{id}')
+                ->controller([BlogApiController::class, 'edit'])
+                ->methods(['PUT'])
+            ;
+        };
 
-.. sidebar:: Routing in Other Formats
+.. tip::
 
-    The ``@Route`` above each method is called an *annotation*. If you'd rather
-    configure your routes in YAML, XML or PHP, that's no problem! Create a new
-    routing file (e.g. ``routes.xml``) in the ``config/`` directory and Symfony
-    will automatically use it.
+    HTML forms only support ``GET`` and ``POST`` methods. If you are calling a
```

Comment:
```suggestion
    HTML forms only support ``GET`` and ``POST`` methods. If you're calling a
```

I'm assuming `you're` is equally understandable to non-native speakers than `you are`. Using `you are` sounds a lot more formal in English - so I'm preferring the more casual tone.